Src/FluentAssertions/Specialized/TaskAssertions.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Specialized
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a task yields the expected result.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class TaskAssertions
+    {
+        private readonly ITimer timer;
+
+        public TaskAssertions(Task subject, ITimer timer)
+        {
+            Subject = subject;
+            this.timer = timer;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Task"/> that is being asserted.
+        /// </summary>
+        public Task Subject { get; private set; }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Task"/> will complete within specified time range.
+        /// </summary>
+        /// <param name="timeSpan">The allowed time span for the operation.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public void CompleteWithin(
+            TimeSpan timeSpan, string because = "", params object[] becauseArgs)
+        {
+            var completed = this.timer.Wait(Subject, timeSpan);
```

Comment:
(minor) You don't need `this` here.

Reply:
I do not want to start such discussion here. I have different understanding for good coding style. I would like to identify members (and I do not line the "_member" aproach).
If you want I can remove "this.".